src/Library/Plugin/Plugin.php

```diff
@@ -333,6 +373,64 @@ public function getSlug() {
 		return $this->slug;
 	}
 
+	/**
+	 * Create page objects based on $this->config parameter
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 */
+	private function setPages() {
+		$pages = [];
+		if ( isset( $this->pluginConfig['pages'] ) ) {
+			foreach ($this->pluginConfig['pages'] as $page ) {
+				$pages[] = new Page( $this, $page );
+			}
+		}
+		$this->pages = $pages;
+	}
+
+	/**
+	 * Get Plugin Pages
+	 *
+	 * @return array
+	 * @access private
+	 * @since SINCEVERSION
+	 */
+	private function getPages() {
+		return $this->pages;
+	}
+
+	/**
+	 * Get Page by Slug
+	 * 
+	 * @param string $slug
+	 * @return Page
+	 * @since SINCEVERSION
+	 */
+	public function getPageBySlug( $slug ) {
+		foreach ( $this->getPages() as $page ) {
+			if ( $page->getSlug() == $slug ) {
+				return $page;
+			}
+		}
+	}
+
+	/**
+	 * Populate $this->pluginConfig from array given in
+	 * constructor
+	 *
+	 * @param array $pluginConfig
+	 * @access private
+	 * @since SINCEVERSION
+	 */
+	private function setPluginConfig( array $pluginConfig ) {
+		if ( $pluginConfig ) {
```

Comment:
Not required, but this method could be a one liner:

`$this->pluginConfig = ! empty( $pluginConfig ) ? $pluginConfig : [];`

Reply:
Done